src/main/scala/nl/knaw/dans/easy/umd/CommandLineOptions.scala

```diff
@@ -0,0 +1,145 @@
+ */
+package nl.knaw.dans.easy.umd
+
+import java.io.File
+import java.net.URL
+
+import com.yourmediashelf.fedora.client.FedoraCredentials
+import nl.knaw.dans.easy.umd.CommandLineOptions.log
+import nl.knaw.dans.easy.umd.{Parameters, Version}
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.rogach.scallop._
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+class CommandLineOptions(args: Array[String] = "-f http:// -fedora-username u -fedora-password p -s s -t t README.md".split(" ")) extends ScallopConf(args) {
+
+  appendDefaultToDescription = true
+  editBuilder(_.setHelpWidth(110))
+
+  printedName = "easy-update-metadata-dataset"
+  val _________ = " " * printedName.length
+
+  version(s"$printedName v${Version()}")
+  banner(
+    s"""
+       |Batch-updates metadata streams in a Fedora Commons repository
+       |
+           |Usage:
+       |
+           |$printedName <synopsis of command line parameters>
+       |${_________} <...possibly continued here>
+       |
+           |Options:
+       |""".stripMargin)
+
+  private val shouldBeFile = singleArgConverter(value =>
+    new File(value) match {
+      case f if f.isFile => f
+      case _ => throw createExecption(s"'$value' is not a file")
+    }
+  )
+  private val shouldBeDir = singleArgConverter(value =>
+    new File(value) match {
+      case f if f.isDirectory => f
+      case _ => throw createExecption(s"'$value' is not a directory")
+    }
+  )
+  private val shouldBeUrl = singleArgConverter(value =>
+    Try {
+      new URL(value)
+    } match {
+      case Success(url) => value
+      case Failure(e) => throw createExecption(s"'$value' is not a valid url: ${e.getMessage}")
+    }
+  )
+
+  private def createExecption(msg: String) = {
+    log.error(msg)
+    new IllegalArgumentException(msg)
+  }
+
+
+  val doUpdate = opt[Boolean](name = "doUpdate", noshort = true,
+    descr = "Without this argument no changes are made to the repository, the default is a test mode that logs the intended changes",
+    default = Some(false))
+
+  val streamID = opt[String](name = "stream-id", short = 's', required = true, descr = "id of fedoara stream to update")
+  val tag = opt[String](name = "tag", short = 't', required = true, descr = "xml tag to change")
+
+  val fedoraUrl = opt[String](name = "fedora-url", short = 'f',
+    descr = "Base url for the fedora repository",
+    default = Some("http://localhost:8080/fedora"))(shouldBeUrl)
+  val fedoraUsername = opt[String](name = "fedora-username", noshort = true,
+    descr = "Username for fedora repository, if omitted provide it on stdin")
+  val fedoraPassword = opt[String](name = "fedora-password", noshort = true,
+    descr = "Password for fedora repository, if omitted provide it on stdin")
+
+  val inputFile = trailArg[File](name = "input-file", required = true, descr = "The CSV file with required changes. Columns: fedoraID, new value, optional old value")(shouldBeFile)
```

Comment:
The optional old value column only applies to https://github.com/DANS-KNAW/easy-update-metadata-fileitem for datasets the query that creates the input does the job.